test/Microsoft.Extensions.Caching.Memory.Tests/EntryLinkExpirationTests.cs

```diff
@@ -94,6 +94,72 @@ public void TokenExpires_LinkedEntry()
         }
 
         [Fact]
+        public void TokenExpires_ParentScopeEntry()
+        {
+            var cache = CreateCache();
+            var obj = new object();
+            string key = "myKey";
+            string key1 = "myKey1";
+            var expirationToken = new TestExpirationToken() { ActiveChangeCallbacks = true };
+
+            using (var link1 = cache.CreateLinkingScope())
+            {
+                cache.Set(key, obj, new MemoryCacheEntryOptions());
+
+                using (var link2 = cache.CreateLinkingScope())
+                {
+                    cache.Set(key1, obj, new MemoryCacheEntryOptions().AddExpirationToken(expirationToken));
+                }
+            }
+
+            Assert.Same(obj, cache.Get(key));
+            Assert.Same(obj, cache.Get(key1));
+
+            expirationToken.Fire();
+
+            object value;
+            Assert.False(cache.TryGetValue(key1, out value));
+            Assert.False(cache.TryGetValue(key, out value));
```

Comment:
Talked with @Tratcher offline, the open question resides in if we want to handle the case where `key` is set before `key1` in this case, or we only need to support when `key` is set after the inner scopes are disposed. If we just care about the later then we don't need the `_entryLink` in the `CacheEntry`.